client/concurrency.go

```diff
@@ -148,26 +147,24 @@ func sendConcurrentRequests(sendFn func() error) error {
 	})
 
 	maybeErrMessage := ""
-	first10Workers := []string{}
 	for i := 0; i < numConReqs; i++ {
 		resp := <-respCh
 		if resp.err != nil {
 			maybeErrMessage += fmt.Sprintf("error #%d: %v\n", i, resp.err)
-		}
-		if i < 10 {
-			first10Workers = append(first10Workers, fmt.Sprintf("Worker #%d", resp.id))
+		} else {
+			log.Printf("Worker #%d done", resp.id)
 		}
 	}
+
 	if maybeErrMessage != "" {
 		return fmt.Errorf("at least one concurrent request failed:\n%s", maybeErrMessage)
 	}
 
-	log.Printf("First 10 workers done:\n%s", strings.Join(first10Workers, "\n"))
-
 	// Validate that the concurrent requests were handled faster than if all
 	// the requests were handled serially, using the single request time
-	// as a benchmark. Some buffer is provided by doubling the single request time.
-	if conReqTime > 2*singleReqTime {
+	// as a benchmark. The concurrent time should be less than half of the
+	// time it would have taken to execute all requests serially.
+	if conReqTime*2 > numConReqs*singleReqTime {
```

Comment:
oh it actually might be better to keep this more strict requirement of 2×singleReqTime